pallets/motion/src/lib.rs

```diff
@@ -0,0 +1,161 @@
+#![cfg_attr(not(feature = "std"), no_std)]
+
+#[cfg(test)]
+mod mock;
+#[cfg(test)]
+mod tests;
+
+#[cfg(feature = "runtime-benchmarks")]
+/// Dispatchable function benchmarks.
+mod benchmarking;
+
+/// Dispatchable function weights.
+pub mod weights;
+pub use pallet::*;
+use sp_runtime::DispatchResult;
+pub use weights::WeightInfo;
+
+extern crate alloc;
+use alloc::{boxed::Box, vec};
+
+/// The motion pallet.
+#[frame_support::pallet]
+pub mod pallet {
+	use frame_support::{
+		dispatch::GetDispatchInfo, pallet_prelude::*, traits::UnfilteredDispatchable,
+	};
+	use frame_system::pallet_prelude::*;
+
+	use super::{DispatchResult, *};
+
+	#[pallet::pallet]
+	pub struct Pallet<T>(_);
+
+	/// Configure the pallet by specifying the parameters and types on which it depends.
+	#[pallet::config]
+	pub trait Config: frame_system::Config {
+		/// The runtime event type.
+		type RuntimeEvent: From<Event<Self>> + IsType<<Self as frame_system::Config>::RuntimeEvent>;
```

Comment:
I think we can get rid of these Runtime* types now by simply adding bounds on the underlying type, removing the need to define them in the runtime. I have done so successfully elsewhere, YMMV tho.

i.e. 
`pub trait Config: frame_system::Config<RuntimeCall: Parameter + UnfilteredDispatchable<RuntimeOrigin = Self::RuntimeOrigin> + GetDispatchInfo>`

Reply:
I think would be a great improvement!
Unfortunately I couldn't make it work. I was running into a cycle:

`cycle detected when computing layout of RuntimeCall`

Reply:
Probably as you were using `Self`

Reply:
I'd l like to revisit this once the changes are merged